tests/test_modeling_tf_common.py

```diff
@@ -438,6 +438,11 @@ def test_pt_tf_model_equivalence(self):
 
     def test_train_pipeline_custom_model(self):
         config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        # head_mask and decoder_head_mask has different shapes than other input args
+        if "head_mask" in inputs_dict:
+            del inputs_dict["head_mask"]
+        if "decoder_head_mask" in inputs_dict:
+            del inputs_dict["decoder_head_mask"]
```

Comment:
Why removing them? We are not supposed to possibly train a model with these arguments?

Reply:
@jplu - Currently, I'm not fully sure how to implement this test with `head_mask` and `decoder_head_mask`, as they have different shapes than `input_ids, attention_mask,...`. While the latter  ones have shapes of `(batch_size, seq_len)`, the `head_mask` has `(num_layers, num_attention_heads)`. This results in the following error
```
> X = tf.data.Dataset.from_tensor_slices(
    (inputs_dict, np.ones((self.model_tester.batch_size, self.model_tester.seq_length, num_labels, 1)))
).batch(1)
----------------------------------
self = Dimension(13), other = Dimension(5)

    def assert_is_compatible_with(self, other):
      """Raises an exception if `other` is not compatible with this Dimension.
    
      Args:
        other: Another Dimension.
    
      Raises:
        ValueError: If `self` and `other` are not compatible (see
          is_compatible_with).
      """
      if not self.is_compatible_with(other):
>       raise ValueError("Dimensions %s and %s are not compatible" %
                         (self, other))
E       ValueError: Dimensions 13 and 5 are not compatible

../../../../../../miniconda3/envs/bart/lib/python3.8/site-packages/tensorflow/python/framework/tensor_shape.py:281: ValueError
```
Do you have any idea of how to overcome this problem?

Furthermore, at this moment, this test does not consider `head_mask` for any BERT-like models because `head_mask` is not tested for these models at all. I'm definitely willing to implement such testing for other than BART-like models as well, once we're sure about the proper implementation.

Reply:
Ok, indeed in a tf dataset all the inputs must share the same shape. To overcome this you should create your own `test_train_pipeline_custom_model` for each Seq2Seq model and replace the tf dataset by the dictionary.

Anyway, you should not modify a common test just to make pass the test for a single model.

Reply:
To be honest, I don't really agree here @jplu . IMO it's fine to remove `head_mask` from the `inputs_dict`. We remove the `head_mask` for other tests as well, such as https://github.com/huggingface/transformers/blob/fac7cfb16a437a97584f6a14c3856b2e06bf0eaa/tests/test_modeling_common.py#L521, https://github.com/huggingface/transformers/blob/fac7cfb16a437a97584f6a14c3856b2e06bf0eaa/tests/test_modeling_common.py#L1076. 

Also the `test_train_pipeline_custom_model` is not meant to test the head masking and removing just means that the test covers the exact same functionalities as before. All "normal" head-masking tests were in enabled in this PR. 
I'm fine with leaving it as it for now

Reply:
Humm, ok so that means you should not be supposed to train BART with such arguments, right?

Reply:
Contemporary research does not consider training models with `head_mask`. The layer masking is applied during inference, once the model is trained with all attention heads.

Reply:
Ok, perfect then! I'm fine to let it like this :)